src/canvasObject.js

```diff
@@ -1,159 +1,128 @@
 'use strict';
 
 require('openseadragon');
+var ImageResource = require('./ImageResource');
 
 var CanvasObject = function(config, dispatcher) {
-  this.fullyOpened = config.fullyOpened || false;
-  this.visible = config.visible || true; // todo: this is not used yet. Do we need it?
+  var self = this;
   this.clipRegion = config.clipRegion;
   this.opacity = config.opacity || 1;
-  this.x = config.x || 0;
-  this.y = config.y || 0;
-  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
   this.index = config.index;
 
   this.id = config.canvas['@id'];
-  this.height = config.canvas.height;
-  this.width = config.canvas.width;
-  this.images = config.canvas.images;
+
+  this.bounds = {
+    x : config.x || 0,
+    y : config.y || 0,
+    height : config.canvas.height,
+    width : config.canvas.width
+  };
+  this.thumbUrl = config.canvas.thumbnail;
+  this.thumbService = config.canvas.images[0].resource.service['@id'];
+
+  // details and alternates possibly go here; disambiguate between them.
+  this.images = config.canvas.images.map(function(image) {
+    return new ImageResource(
+      {
+        tileSource: image.resource.service['@id'] + '/info.json'
+      },
+      self,
+      dispatcher
+    );
+  });
+
   this.label = config.canvas.label;
-  this.tileSourceUrl = config.canvas.images[0].resource.service['@id'] + '/info.json';
   this.viewingHint = config.canvas.viewingHint;
 
   this.dispatcher = dispatcher;
 };
 
 CanvasObject.prototype = {
-  openTileSource: function(viewer) {
-    var self = this;
+  openTileSource: function(viewer, imageIndex) {
```

Comment:
I don't think it's appropriate to pass `viewer` into here...the CanvasObject should just know its viewer, so the caller doesn't have to keep reminding it. If we don't know the viewer when the CanvasObject is constructed, maybe we should have a setViewer function for once the viewer has been created.